components/ShoppingListEditor/ShoppingListEditor.js

```diff
@@ -86,22 +85,28 @@ export default function ShoppingListEditor({ items, onDelete, onAdd }) {
           maxLength="30"
           ref={inputRef} //set ref to set autofocus after submit
           value={itemInput}
-          onInput={(event) => setItemInput(event.target.value)} //don't use onChange() -> it ignores some events!!!
-          onFocus={triggerDropDown}
-          onBlur={() => setDropDownReset(true)} //to close drop down, when losing focus
-        />
-        <InputDropDown
-          optionElements={dropDownItems}
-          ariaLabel="add item"
-          onAdd={onAdd}
+          onInput={(event) => handleItemInput(event)} //don't use onChange() -> it ignores some events!!!
+          onFocus={() => triggerDropDown(itemInput)}
+          //close drop down, when losing focus
```

Comment:
what exactly happens here if I focus the input but don't type anything yet? I guess the dropdown won't be displayed since the dropDownItems.length is still 0. I guess you don't even need to check if the input is focussed, since the dropdownItems are displayed as soon as the array gets some elements. This happens when you write something in the input, for that the input needs to be focused anyway. Check if it still works when you leave out this event listener.